tests/beaker_tests/cisco_interface_channel_group/test_interface_channel_group.rb

```diff
@@ -0,0 +1,187 @@
+#                        This key overrides a tests[:platform] key
+# tests[id][:desc] - a string to use with logs & debugs
+# tests[id][:manifest] - the complete manifest, as used by test_harness_common
+# tests[id][:resource] - a hash of expected states, used by test_resource
+# tests[id][:resource_cmd] - 'puppet resource' command to use with test_resource
+# tests[id][:ensure] - (Optional) set to :present or :absent before calling
+# tests[id][:code] - (Optional) override the default exit code in some tests.
+#
+# These keys are local use only and not used by test_harness_common:
+#
+# tests[id][:manifest_props] - This is essentially a master list of properties
+#   that permits re-use of the properties for both :present and :absent testing
+#   without destroying the list
+# tests[id][:resource_props] - This is essentially a master hash of properties
+#   that permits re-use of the properties for both :present and :absent testing
+#   without destroying the hash
+# tests[id][:title_pattern] - (Optional) defines the manifest title.
+#
+intf = 'ethernet1/1'
+tests['default_properties'] = {
+  desc:           '1.1 Default Properties',
+  title_pattern:  intf,
+  code:           [0, 2],
+  manifest_props: {
+    channel_group: 'default',
+    description:   'default',
+    shutdown:      'default',
+  },
+  resource:       {
+    'channel_group' => 'false',
+    'shutdown'      => 'true',
+  },
+}
+
+tests['non_default_properties'] = {
+  desc:           '2.1 Non Default Properties commands',
+  title_pattern:  intf,
+  manifest_props: {
+    channel_group: 201,
+    description:   'chan group desc',
+    shutdown:      'false',
+  },
+  resource:       {
+    'channel_group' => '201',
+    'description'   => 'chan group desc',
+    'shutdown'      => 'false',
+  },
+}
+
+# Create actual manifest for a given test scenario.
+def build_manifest_interface_channel_group(tests, id)
+  manifest = prop_hash_to_manifest(tests[id][:manifest_props])
+  if tests[id][:ensure] == :absent
+    state = 'ensure => absent,'
+    tests[id][:resource] = { 'ensure' => 'absent' }
+  else
+    state = 'ensure => present,'
+  end
+
+  tests[id][:title_pattern] = id if tests[id][:title_pattern].nil?
+  tests[id][:manifest] = "cat <<EOF >#{PUPPETMASTER_MANIFESTPATH}
+  \nnode default {
+  cisco_interface_channel_group { '#{tests[id][:title_pattern]}':
+    #{state}\n#{manifest}
+  }\n}\nEOF"
+
+  cmd = PUPPET_BINPATH +
+        "resource cisco_interface_channel_group '#{tests[id][:title_pattern]}'"
+  tests[id][:resource_cmd] = get_namespace_cmd(agent, cmd, options)
+end
+
+# Wrapper for interface_channel_group specific settings prior to calling the
+# common test_harness.
+def test_harness_interface_channel_group(tests, id)
+  return unless platform_supports_test(tests, id)
+
+  tests[id][:ensure] = :present if tests[id][:ensure].nil?
+
+  # Workaround for (ioctl) facter bug on n7k ***
+  tests[id][:code] = [0, 2] if platform[/n7k/]
```

Comment:
Not sure why you need this change?  I have the same issue on my 5|7k but did not need to make this change in my beaker tests...